tests/integration/integration_util_test.go

```diff
@@ -453,3 +458,132 @@ func printfTestValidation(t *testing.T, stack integration.RuntimeValidationStack
 	assert.Equal(t, 11, foundStdout)
 	assert.Equal(t, 11, foundStderr)
 }
+
+// testLogWriter is an io.Writer
+// that writes to the provided testing.T.
+//
+// Use newTestLogWriter to ensure it is flushed
+// of any input when the test finishes.
+type testLogWriter struct {
```

Comment:
This can likely be re-used elsewhere.
I will defer moving this somewhere shared for when we need it.